app/Contact.php

```diff
@@ -229,6 +229,9 @@ public function setVcard4($vcard)
         $this->url = !empty($vcard->url->uri)
             ? (string)$vcard->url->uri
             : null;
+        if (!$this->url && $vcard->impp->uri) {
```

Comment:
I tend to add a carriage return before if() for readability.
Also testing `$this->url == null` seems cleaner :)